### DISCUSSION TITLE
Convert user to id

### DISCUSSION THREAD

**Discussion by CubixFR** [discordjs/discord.js#5307] [Q&A] [answered]
Hello,
I would like to get the id of a user for a system that uses ids. The problem is, I can't find any line of code that can do this. I know there is a way to do the reverse (Id -> User), but that's not what I'm interested in.
Thank you in advance.

**Comment by Androz2091** [comment]
Hello, you can use `message.mentions.users.first().id`.

**Comment by CubixFR** [comment]
I am using this `${message.mentions.users.first(1)}` but its gives me the username and not the id and that's why I can't use `user.id`.

**Reply by Androz2091** [reply]
I updated my answer

**Reply by CubixFR** [reply]
That answers me `undefined`

EDIT: It works Thanks

**Reply by Androz2091** [reply]
perfect 👌 
if you're still here, note that you can mark this issue as resolved

**Comment by kyranet** [comment, marked as answer]
**User -> ID**: [`user`](https://discord.js.org/#/docs/main/stable/class/User).[`id`](https://discord.js.org/#/docs/main/stable/class/User?scrollTo=id).
**ID -> User**: [`client`](https://discord.js.org/#/docs/main/stable/class/Client).[`users`](https://discord.js.org/#/docs/main/stable/class/Client?scrollTo=users).[`cache`](https://discord.js.org/#/docs/main/stable/class/UserManager?scrollTo=cache).[`get`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get)([`id`](https://discord.js.org/#/docs/main/stable/typedef/Snowflake));
